File: 0728-self-dividing-numbers/0728-self-dividing-numbers.java
class Solution {
    public boolean isSelfDividingNumber(int number){
        int tempNumber = number;
        while(tempNumber > 0){
            int digit = tempNumber % 10;
            if(digit == 0 || number%digit != 0) return false;
            tempNumber/=10;
        }
        return true;
    }

    public List<Integer> selfDividingNumbers(int left, int right) {
        List<Integer> selfDividingNums = new ArrayList<Integer>();
        while(left<=right){
            if(isSelfDividingNumber(left))
                selfDividingNums.add(left);
            left++;
        }
        return selfDividingNums;
    }
}

File: 0728-self-dividing-numbers/0728-self-dividing-numbers.js
/**
 * @param {number} left
 * @param {number} right
 * @return {number[]}
 */

var isSelfDividing = function(number){
  if(number < 10) return true;
  const digits = new Set(number.toString().split(""));
  for(const digit of digits){
    if(digit === '0' || number % Number(digit) !== 0) return false;
  }
  return true;
}

var selfDividingNumbers = function(left, right) {
   const result = [];
    const parseNum = (num) => {
        let copyNum = num;

        while (num > 0) {
            if (copyNum % (num % 10) !== 0) {
                return false;
            }

            num = Math.floor(num / 10);
        }

        return true;
    }

    while (left <= right) {
        const isFit = parseNum(left);

        if (isFit) {
            result.push(left);
        }

        left++;
    }

    return result;
};